.github/workflows/ci.yml

```diff
@@ -0,0 +1,40 @@
+name: CI
+
+on:
```

Comment:
My understanding of this is that any commit to master or pr to master will initiate a push to ECR. Was there a plan to push images for any commit on any branch such that dev branches/features could be referenced in the step-functions / pipeline as well?

Reply:
so, this is something we have spoken about a few times in the context of multiple repos now. You're right that at the moment we will only publish things on merge to main, and that it may be useful to publish more artifacts. The things we need to consider:

- only publishing things on main is simple, and the semantics of `latest` are easy to define, we'd need a plan to handle multiple branches in a sensible fashion
- how do we make sure we clean up branch tags, but maintain `main` release versions for longer...?
- what's the implication in terms of storage?

Definitely something to look into post-COP. Perhaps a tech debt ticket needs to be created in linear?

Reply:
oh, and this script also runs on any PR opened with `main` as the target, so there's no technical blocker to implementing the above